packages/splitter/src/useSplitter.ts

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { composeEventHandlers } from '@zendeskgarden/container-utilities';
+import React, { HTMLProps, MouseEvent, useCallback, useEffect, useRef, useState } from 'react';
+
+export enum SplitterOrientation {
+  HORIZONTAL = 'horizontal',
+  VERTICAL = 'vertical'
+}
+
+export enum SplitterType {
+  FIXED = 'fixed',
+  VARIABLE = 'variable'
+}
+export interface IUseSplitterProps extends Omit<HTMLProps<any>, 'onChange' | 'controls'> {
+  /** A label for the separator */
+  label?: string;
+  /** An id identifying the element labeling the separator */
+  labelledBy?: string;
```

Comment:
The existing packages tend to pick a path on `aria-label` vs `aria-labelledby` based on the items under immediate control. In this case, I would suggest limiting to `aria-label` with treatment similar to what we see [here](https://github.com/zendeskgarden/react-containers/blob/b46d1743ac0d19a40e54ad587845960a061db7e0/packages/pagination/src/usePagination.ts#L56-L87).

If a consumer really wants a different setup, we'll let them spread the props they want under the `Splitter` element TBD under `react-components` for an alternate a11y experience.